libkbfs/kbfs_ops.go

```diff
@@ -442,9 +455,21 @@ func (fs *KBFSOpsStandard) getOrInitializeNewMDMaster(ctx context.Context,
 		return false, ImmutableRootMetadata{}, tlf.NullID, err
 	}
 
-	md, err = mdops.GetForTLF(ctx, h.tlfID, nil)
-	if err != nil {
-		return false, ImmutableRootMetadata{}, tlf.NullID, err
+	if rev, isRevBranch := fb.Branch.HasRevision(); isRevBranch {
+		fs.log.CDebugf(ctx, "Getting archived revision %d for branch %s",
+			rev, fb.Branch)
+		md, err = getSingleMD(
+			ctx, fs.config, h.tlfID, kbfsmd.NullBranchID, rev,
+			kbfsmd.Merged, nil)
+		// This will error if there's no corresponding MD.
+		if err != nil {
+			return false, ImmutableRootMetadata{}, tlf.NullID, err
+		}
```

Comment:
I think I'd rather have an explicit `return false, md, h.tlfID, nil` here, rather than falling through. Then you don't need the comment above.

Reply:
I think returning an error is what we want here -- otherwise someone who looks up `.kbfs_archived_rev=10000`, when the current head is only 9000, will get an error instead of it just looking empty.  I can clarify that in the comment.

Reply:
I meant after the error check and return.